internal/cloud/gcp/metadata.go

```diff
@@ -83,7 +83,18 @@ func (m *Metadata) Self(ctx context.Context) (metadata.InstanceMetadata, error)
 	if err != nil {
 		return metadata.InstanceMetadata{}, err
 	}
-	return m.api.RetrieveInstance(ctx, project, zone, instanceName)
+	subnetCIDR, err := m.api.RetrieveSubnetworkAliasCIDR(ctx, project, zone, instanceName)
+	if err != nil {
+		return metadata.InstanceMetadata{}, err
+	}
+
+	instance, err := m.api.RetrieveInstance(ctx, project, zone, instanceName)
+	if err != nil {
+		return metadata.InstanceMetadata{}, err
+	}
+	instance.SecondaryIPRange = subnetCIDR
+
+	return instance, nil
```

Comment:
Is the `SecondaryIPRange` information only relevant for listing `Self`, or would we also want this information when calling `List` or `GetInstance`?

Reply:
Good point. I've only added it to `Self` since I know that this is the only place where it is consumed from. This is of course not a very good reason of only filling it that case. I think from an API/Interface point of view it makes sense to always fill it where possible. Having said that, I'd push this discussion/change to the general metadata refactoring.

Reply:
I can add it to `List` and `GetInstance` :+1: